src/ResourceStore.php

```diff
@@ -0,0 +1,237 @@
+<?php
+
+namespace silverorange;
+
+use GuzzleHttp\Psr7\Request;
+use GuzzleHttp\Client as HttpClient;
+
+class ResourceStore
+{
+    // {{{ protected properties
+
+    protected $json_api_base;
+    protected $json_api_headers;
+
+    protected $http_client;
+
+    protected $class_by_type = array();
+
+    protected $resources = array();
+
+    // }}}
+    // {{{ public function __construct()
+
+    public function __construct($json_api_base, $json_api_headers = [])
+    {
+        $this->json_api_base = $json_api_base;
+        $this->json_api_headers = $json_api_headers;
+
+        $this->initHttpClient();
+    }
+
+    // }}}
+    // {{{ protected function initHttpClient()
+
+    protected function initHttpClient()
+    {
+        $default_headers = $this->getDefaultHttpHeaders();
+
+        $this->http_client = new HttpClient(
+            [
+                'base_uri' => $this->json_api_base,
+                'headers' => array_merge(
+                    $default_headers,
+                    $this->json_api_headers
+                ),
+            ]
+        );
+    }
+
+    // }}}
+    // {{{ protected function getDefaultHttpHeaders()
+
+    protected function getDefaultHttpHeaders()
+    {
+        return [
+            'Accept' => 'application/vnd.api+json',
+            'Content-Type' => 'application/vnd.api+json',
+        ];
+    }
+
+    // }}}
+    // {{{ public function addClass()
+
+    public function addClass($type, $class)
```

Comment:
I'll remove them from the PR. These methods will probably be added back in the future so that we can map a PHP class to a type and that will allow us to make new objects instead of just loading existing ones.